### PR TITLE
feat(js): added ability to register toggleable menu items server side

### LINKED ISSUE #8158: add a way to add toggle menu items from php / output/url states
We currently have this js:
'elgg.ui.registerTogglableMenuItems('feature', 'unfeature');'

can't we come up with a way to autodetect those with a generic js function? or register the menu items as toggleable from php?

This could prevent adding js files or have inline js for just toggleable menu items.

Any preference?

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #8158

### CODE REVIEW

**Comment by jdalsem** [comment]
Does not break current way of registering toggleable menu items and does not add intelligence to MenuItem classes

File: docs/guides/menus.rst [hunk 1]
```diff
@@ -256,7 +256,35 @@ with a high degree of control and flexibility when styling the site.
 		<li class="elgg-menu-item elgg-menu-item-bar"></li>
 	</ul>
 
+Toggling Menu Items
+===================
+
+There are situations where you wish to toggle menu items that are actions that are the opposite
+of each other and ajaxify them. E.g. like/unlike, friend/unfriend, ban/unban, etc. Elgg has built-in support
```
Comment (mrclay) [inline docs/guides/menus.rst]: Likes doesn't work this way anymore and I certainly wouldn't recommend anyone build on the old model.

Comment (jdalsem) [reply docs/guides/menus.rst]: Likes is only mentioned as an example. I am not suggesting to build on any model other than documenting Elgg has built in toggle features

Comment (mrclay) [reply docs/guides/menus.rst]: Fair point and this definitely is cleaner than the status quo.

**Review by mrclay — APPROVED** [review]

**Pull request merged**